v-next/hardhat/src/internal/builtin-plugins/solidity-test/stack-trace-solidity-errors.ts

```diff
@@ -46,13 +46,15 @@ export function getMessageFromLastStackTraceEntry(
     case StackTraceEntryType.DIRECT_LIBRARY_CALL_ERROR:
       return `Library was called directly`;
 
+    /* These types are not associate with a more detailed error message */
     case StackTraceEntryType.UNRECOGNIZED_CREATE_ERROR:
     case StackTraceEntryType.UNRECOGNIZED_CONTRACT_ERROR: {
-      return "Contract reverted without a reason string";
+      return undefined;
```

Comment:
To check I have understood. For these error types, we now fallback to the reason that is passed from EDR (rather than a hard coded string on the Hardhat side). That fallback is managed by passing undefined which indicates we should read the error message from the EDR side?

Reply:
Yes, that's correct!